monitoring/prometheusalerts.md

```diff
@@ -0,0 +1,87 @@
+
+## Create some alerts
+
+We have an [example alerts file](prometheusalerts.yml) :
+
+```
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: camel-alerts
+  namespace: fuse7
+spec:
+  groups:
+  - name: CamelAlerts
+    rules:
+    - alert: ExchangesFailed
+      expr: org_apache_camel_ExchangesFailed{job="spring-boot-camel"} > 0
+      for: 1m
+      labels:
+        severity: critical
+      annotations:
+        summary: "{{$labels.job}} exchange failure in context {{$labels.context}}"
+        description: "{{$labels.job}} exchange failed - context {{$labels.context}}, namespace {{$labels.namespace}}, processor {{$labels.processor}}"
+    - alert: MeanProcessingTimeAboveValueForTenMinutes
+      expr: org_apache_camel_MeanProcessingTime{job="spring-boot-camel"} >= 5000
+      for: 10m
+      labels:
+        severity: critical
+      annotations:
+        summary: "{{$labels.job}} mean processing time was {{ $value }}, above specified value of 5000"
+        description: "{{$labels.job}} mean processing time was {{ $value }}, above specified value of 5000"
+    - alert: CamelContextDownAfterTenMinutes
+      expr: org_apache_camel_StartedRoutes{job="spring-boot-camel"} == 0
+      for: 10m
+      labels:
+        severity: critical
+      annotations:
+        summary: "{{$labels.job}} has no started routes after ten minutes"
+        description: "{{$labels.job}} has {{ $value }} started routes after ten minutes"
+
+```
+
+If you were to use the above example, there are a few things that need to be set.   `job="spring-boot-camel` refers to the Prometheus ServiceMonitor that you created in the first section.   To find what service monitors you've already created in your namespace :
+
+```
+$ oc get ServiceMonitor
+NAME                AGE
+spring-boot-camel   2d1h
+```
+
+Edit the file above to use your namespace (change the `namespace:` line) and change the `job=""` to reflect the name of your ServiceMonitor.    Then apply your Prometheus alerts :
+
+```
+$ oc create -f prometheusalerts.yml 
+prometheusrule.monitoring.coreos.com/camel-alerts created
+```
+
+
+## Using Thanos to view your Alerts
+
+User Workload Monitoring includes the thanos-ruler which gives us a view into the state of your alerts.
+
+Get the route for Thanos from openshift-user-workload-monitoring:
+
+```
+$ oc get routes -n openshift-user-workload-monitoring
+NAME           HOST/PORT                                                                                  PATH   SERVICES       PORT   TERMINATION          WILDCARD
+thanos-ruler   thanos-ruler-openshift-user-workload-monitoring.apps.promtest.local          thanos-ruler   web    reencrypt/Redirect   None
+```
+
+Use the route listed and plug it into your browser :
+
+![Thanos Alerting](https://github.com/jboss-fuse/application-templates/raw/master/monitoring/img/thanosalerts.png)
```

Comment:
Is there a reason to use an absolute path?  If not we can probably be shorter like this:
```
![Thanos Alerting](img/thanosalerts.png)
```

Reply:
Fixed!